### PR TITLE
Combine the two types.go file into one.

### PR DESCRIPTION
**Pull request by yizhang4321**

### CODE REVIEW

File: pkg/agent/datamodel/types.go [hunk 1]
```diff
@@ -1286,3 +1293,395 @@ func FormatProdFQDNByLocation(fqdnPrefix string, location string, cloudSpecConfi
...
+	CloudSpecConfig               *AzureEnvironmentSpecConfig
+	K8sComponents                 *K8sComponents
+	AgentPoolProfile              *AgentPoolProfile
+	TenantID                      string
+	SubscriptionID                string
+	ResourceGroupName             string
+	UserAssignedIdentityClientID  string
+	ConfigGPUDriverIfNeeded       bool
+	EnableGPUDevicePluginIfNeeded bool
+	EnableDynamicKubelet          bool
```
Comment (xuto2) [inline pkg/agent/datamodel/types.go:1323]: @bowang-666  FYI potential conflict with your change that renames this property.

Comment (bowang-666) [reply pkg/agent/datamodel/types.go:1323]: Just merged this https://github.com/Azure/AgentBaker/pull/411
Could you help rename property name here?

Comment (yizhang4321) [reply pkg/agent/datamodel/types.go]: I have resolved merge conflict with #411. Please take another look.

**Pull request merged**